pkg/tasks/c1api/service_client.go

```diff
@@ -0,0 +1,253 @@
+package c1api
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"math"
+	"net"
+	"os"
+	"time"
+
+	v1 "github.com/conductorone/baton-sdk/pb/c1/connectorapi/baton/v1"
+	"github.com/conductorone/baton-sdk/pkg/sdk"
+	"github.com/conductorone/baton-sdk/pkg/ugrpc"
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+const (
+	// API allows for up to 4MB chunks, but that is also the max gRPC message size. We use 3MB to be safe.
+	fileChunkSize = 3 * 1024 * 1024 // 3MB
```

Comment:
we changed this to 1mb i think on the c1 side?

Reply:
This is lowered to 512KB now.